app/cdap/components/SourceControlManagement/LocalPipelineListView/index.tsx

```diff
@@ -0,0 +1,163 @@
+import {
+  countPushFailedPipelines,
+  getNamespacePipelineList,
+  pushSelectedPipelines,
+  reset,
+  resetPushStatus,
+  setLoadingMessage,
+  setLocalPipelines,
+  setNameFilter,
+  toggleCommitModal,
+  toggleShowFailedOnly,
+} from '../store/ActionCreator';
+import { SearchBox } from '../SearchBox';
+import LoadingSVGCentered from 'components/shared/LoadingSVGCentered';
+import T from 'i18n-react';
+import { CommitModal } from './CommitModal';
+import cloneDeep from 'lodash/cloneDeep';
+import styled from 'styled-components';
+import PrimaryTextButton from 'components/shared/Buttons/PrimaryTextButton';
+import { LocalPipelineTable } from './PipelineTable';
+import { useOnUnmount } from 'services/react/customHooks/useOnUnmount';
+
+const PREFIX = 'features.SourceControlManagement.push';
+
+const StyledSelectionStatusDiv = styled.div`
+  display: flex;
+  text-align: center;
+  margin: 10px 2px;
+
+  div {
+    margin-right: 10px;
+    padding-top: 4px;
+  }
+`;
+
+export const LocalPipelineListView = () => {
+  const {
+    ready,
+    localPipelines,
+    nameFilter,
+    selectedPipelines,
+    commitModalOpen,
+    loadingMessage,
+    showFailedOnly,
+  } = useSelector(({ push }) => push);
+
+  const pushFailedCount = countPushFailedPipelines();
+
+  useEffect(() => {
+    if (!ready) {
+      getNamespacePipelineList(getCurrentNamespace(), nameFilter);
+    }
+  }, [ready]);
+
+  useOnUnmount(() => reset());
+
+  const onPushSubmit = (commitMessage: string) => {
+    resetPushStatus();
+    const pushedPipelines = cloneDeep(localPipelines);
+    const payload = {
+      commitMessage,
+    };
+    toggleCommitModal();
+    pushSelectedPipelines(
+      getCurrentNamespace(),
+      selectedPipelines,
+      payload,
+      setLoadingMessage
+    ).subscribe({
+      next(res: any) {
+        const currentPipeline = pushedPipelines.find((pipeline) => pipeline.name === res.name);
+        if (res.statusCode !== 200) {
+          currentPipeline.error = res.message;
+          currentPipeline.success = false;
+        } else {
+          currentPipeline.success = true;
+        }
+      },
+      complete() {
+        setLocalPipelines(pushedPipelines);
+        setLoadingMessage(null);
+      },
+    });
+  };
+
+  return (
+    <div style={{ height: 'calc(100% - 150px)' }}>
+      <SearchBox nameFilter={nameFilter} setNameFilter={setNameFilter} />
+      {selectedPipelines.length ? (
+        <StyledSelectionStatusDiv>
+          <div>
+            {T.translate(`${PREFIX}.pipelinesSelected`, {
+              selected: selectedPipelines.length,
+              total: localPipelines.length,
+            })}
+          </div>
+          {pushFailedCount ? (
+            <>
+              <div style={{ color: '#d15668' }}>
+                {T.translate(`${PREFIX}.pipelinesPushedFail`, {
+                  count: pushFailedCount,
+                })}
+              </div>
+              <PrimaryTextButton onClick={toggleShowFailedOnly}>
```

Comment:
Should this be a [switch](https://v4.mui.com/components/switches/)? There's also the [Toggle Button](https://v4.mui.com/components/toggle-button/), which is in labs in v4 but in core for v5. Or have another way of visually indication the toggle state?

Reply:
borrowed this design from replication feature. But yes I can discuss it with Arunima